src/librustc_mir/interpret/eval_context.rs

```diff
@@ -291,22 +298,35 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         ty.is_freeze(*self.tcx, self.param_env, DUMMY_SP)
     }
 
+    pub(super) fn subst_and_normalize_erasing_regions_in_frame<T: TypeFoldable<'tcx>>(
+        &self,
+        value: T,
+    ) -> T {
+        // HACK(oli-obk): see `self.substs` docs
+        let substs = self.stack.last().map_or(self.substs, |frame| frame.instance.substs);
+        self.subst_and_normalize_erasing_regions(substs, value)
+    }
+
     pub(super) fn subst_and_normalize_erasing_regions<T: TypeFoldable<'tcx>>(
```

Comment:
When should one call this instead of `subst_and_normalize_erasing_regions_in_frame`? Should existing callers be ported?

Both methods should have doc comments.

Reply:
documented